addons/account/models/partner.py

```diff
@@ -565,6 +569,28 @@ def _compute_show_credit_limit(self):
         for partner in self:
             partner.show_credit_limit = self.env.company.account_use_credit_limit
 
+    api.depends('credit', 'total_invoiced')
+    def _compute_days_sales_outstanding(self):
+        for partner in self:
+            partner_and_children = [child for child in partner.child_ids]
+            partner_and_children.append(partner)
+            combined_invoices = [invoice
+                                 for partner_id in partner_and_children
+                                 for invoice in partner_id.invoice_ids
+                                 if invoice.invoice_date and
+                                    invoice.is_sale_document(include_receipts=True) and
+                                    invoice.state not in ['cancel', 'draft'] and
+                                    invoice.company_id == self.env.company
+                                 ]
+
+            if not combined_invoices or not partner.total_invoiced:
+                partner.days_sales_outstanding = 0
+                continue
+
+            oldest_invoice = min(combined_invoices, key=lambda x: x.invoice_date)
```

Comment:
It seems like `combined_invoices` is used to find the oldest invoice.
this seems to duplicate the logic in `_invoice_total`...
will it not be better if they both shared the same function `_partner_invoices_domain`: `_invoice_totals` will use it for a `read group`, whereas you will use it with `limit=1, order="date DESC"`

(I'm not sure it will work considering `_invoice_total` uses `account.invoice.report` - might be worth a try though - to prevent discrepancies in the future)
perhaps @qdp-odoo has an opinion on this